libminifi/include/core/repository/VolatileRepository.h

```diff
@@ -331,22 +132,240 @@ class VolatileRepository : public core::Repository, public std::enable_shared_fr
     else
       return false;
   }
-  /**
-   * Purges the volatile repository.
-   */
-  void purge();
 
- private:
   std::map<std::string, std::shared_ptr<minifi::Connection>> connectionMap;
-
-  std::atomic<uint32_t> current_size_;
+  // current size of the volatile repo.
+  std::atomic<size_t> current_size_;
+  // current index.
   std::atomic<uint16_t> current_index_;
-  std::vector<AtomicEntry*> value_vector_;
+  // value vector.
```

Comment:
i thought you were using a map to store the key and resource claim, why we need the value_vector_ here